agent/agentbootstrap/bootstrap.go

```diff
@@ -201,22 +204,49 @@ func (b *AgentBootstrap) Initialize(ctx stdcontext.Context) (_ *state.Controller
 
 	stateParams := b.stateInitializationParams
 
+	dqliteBootstrapOps := []database.BootstrapOpt{
+		ccbootstrap.InsertInitialControllerConfig(stateParams.ControllerConfig),
+		cloudbootstrap.InsertCloud(stateParams.ControllerCloud),
+	}
+
 	// Add the controller model cloud and credential to the database.
 	cloudCred, cloudCredTag, err := b.getCloudCredential()
 	if err != nil {
 		return nil, errors.Annotate(err, "getting cloud credentials from args")
 	}
-	ctrlCloud := stateParams.ControllerCloud
-	ctrlCloud.IsControllerCloud = true
+	if cloudCredTag.Id() != "" {
```

Comment:
There should always be a credential for the controller cloud - this should never be empty.

Reply:
This isn't the conversation we had in London. I can fix up.